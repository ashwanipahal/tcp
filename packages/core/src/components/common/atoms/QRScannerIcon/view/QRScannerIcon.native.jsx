import React from 'react';
import CustomIcon from '../../Icon';
import { ICON_NAME, ICON_FONT_CLASS } from '../../Icon/Icon.constants';

const onQRIconFocus = (navigation, closeModal) => {
  if (navigation) {
    navigation.navigate('QRScanner');
  }
  if (closeModal) {
    closeModal();
  }
};

const QRScannerIcon = ({ navigation, closeModal, style, size }) => {
  return (
    <CustomIcon
      onPress={() => onQRIconFocus(navigation, closeModal)}
      iconFontName={ICON_FONT_CLASS.Icomoon}
      name={ICON_NAME.qrcode}
      size={size}
      color="gray.600"
      accessibilityLabel="barcode"
      style={style}
      isButton
    />
  );
};

export default QRScannerIcon;
