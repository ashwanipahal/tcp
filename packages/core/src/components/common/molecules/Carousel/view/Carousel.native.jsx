import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import Carousel from 'react-native-snap-carousel';
import { Image } from '../../../atoms';
import config from '../config';
import theme from '../../../../../../styles/themes/TCP';

const playIcon = require('../../../../../../assets/play.png');
const pauseIcon = require('../../../../../../assets/pause.png');

// @flow
type Props = {
  carouselConfig: Object,
  data: Array<Object>,
  renderItem: Function,
  onSnapToItem: Function,
  width: Number,
  height: Number,
  slideStyle: Object,
};

type State = {
  autoplay: Boolean,
};

/**
 * Default settings for Carousel.
 */
const defaults = { ...config.CAROUSEL_MOBILE_DEFAULTS };

/**
 * Style for play pause icons.
 */
const { colors, zindex } = theme;
const { playIconHeight, playIconWidth } = config.CAROUSEL_MOBILE_CONFIG;
const touchableStyle = {
  backgroundColor: colors.WHITE,
  borderRadius: 15,
  position: 'absolute',
  bottom: 12,
  left: '50%',
  height: playIconHeight,
  width: playIconWidth,
  transform: [
    {
      translateX: -15,
    },
  ],
  zIndex: zindex.zOverlay,
};

/**
 * @function Carousel component that creates carousel using
 * third party library 'react-native-snap-carousel'
 */
class SnapCarousel extends React.PureComponent<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      autoplay: true,
    };
    this.getPlayButton = this.getPlayButton.bind(this);
    this.play = this.play.bind(this);
    this.pause = this.pause.bind(this);
  }

  /**
   * @function getPlayComponent function gets DOM reference of slider component.
   * @param {[Object]} element [Event object of click].
   * @return {node} function returns slider element.
   */
  getPlayButton() {
    const { autoplay } = this.state;
    return autoplay ? (
      <TouchableOpacity accessibilityRole="button" onPress={this.pause} style={touchableStyle}>
        <Image source={pauseIcon} height={playIconHeight} width={playIconWidth} />
      </TouchableOpacity>
    ) : (
      <TouchableOpacity accessibilityRole="button" onPress={this.play} style={touchableStyle}>
        <Image source={playIcon} height={playIconHeight} width={playIconWidth} />
      </TouchableOpacity>
    );
  }

  /**
   * @function play function enable autoplay for carousel
   * also update component state.
   */
  play() {
    this.carousel.startAutoplay();
    this.togglePlay();
  }

  /**
   * @function pause function stops/pause autoplay for carousel
   * also update component state.
   */
  pause() {
    this.carousel.stopAutoplay();
    this.togglePlay();
  }

  /**
   * @function togglePlay function updates component state
   * after each tap of play pause button.
   */
  togglePlay() {
    const { autoplay } = this.state;
    this.setState({ autoplay: !autoplay });
  }

  render() {
    const {
      carouselConfig: { autoplay: defaultAutoplay },
      data,
      height,
      width,
      onSnapToItem,
      renderItem,
      slideStyle,
    } = this.props;

    return (
      <View>
        <Carousel
          ref={c => {
            this.carousel = c;
          }}
          onSnapToItem={onSnapToItem}
          data={data}
          renderItem={renderItem}
          sliderWidth={width}
          itemWidth={width}
          sliderHeight={height}
          itemHeight={height}
          slideStyle={slideStyle}
          {...defaults}
        />
        {defaultAutoplay && this.getPlayButton()}
      </View>
    );
  }
}

export default SnapCarousel;
