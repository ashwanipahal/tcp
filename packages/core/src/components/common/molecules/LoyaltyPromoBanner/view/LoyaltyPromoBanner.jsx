import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';
import { Anchor, RichText, Row, Col } from '../../../atoms';
import { readCookie, setCookie } from '../../../../../utils/cookie.util';
import { loyalityAnalyticsValue } from '../../../../../constants/analytics';
import withStyles from '../../../hoc/withStyles';

import style from '../LoyaltyPromoBanner.style';

/**
 * Check if required UUID cookie available else return default.
 */
const getUUID = uuidCookieString => {
  const UUID = readCookie(uuidCookieString);
  return UUID ? UUID.split(',')[0] : '-1002';
};

/* istanbul ignore next */
const LoyaltyPromoBanner = props => {
  const {
    className,
    cookieID,
    richTextList: [{ richText, link }],
    dataLocator,
  } = props;
  const cookieName = `${cookieID}_${getUUID('WC_USERACTIVITY_')}`;
  const [bannerClosed, setBannerClosed] = useState(true);

  useEffect(() => {
    if (!readCookie(cookieName)) {
      setBannerClosed(false);
    }
  }, [bannerClosed]);

  const closeButtonHandler = () => {
    const currentDate = new Date();
    setCookie({
      key: cookieName,
      value: currentDate.toGMTString(),
      daysAlive: 10,
    });
    setBannerClosed(true);
  };

  return bannerClosed ? null : (
    <div className={`${className} content-wrapper`}>
      <Row>
        <Col
          colSize={{
            small: 6,
            medium: 8,
            large: 12,
          }}
        >
          {link ? (
            <ClickTracker
              as={Anchor}
              to={link.url}
              asPath={link.url}
              target={link.target}
              title={link.title}
              dataLocator={dataLocator || `loyalty-promo-banner`}
              clickData={{
                customEvents: ['event80', 'event81'],
                internalCampaignId: loyalityAnalyticsValue,
              }}
            >
              <RichText richTextHtml={richText.text} />
            </ClickTracker>
          ) : (
            <RichText richTextHtml={richText.text} />
          )}
          <button
            aria-label="close"
            className="loyalty-promo-close-btn"
            onClick={closeButtonHandler}
          >
            <svg
              className="loyalty-promo-close-btn-icon"
              viewBox="0 0 25 25"
              aria-hidden="true"
              role="img"
            >
              <path
                fill="#a0a0a0"
                fillRule="nonzero"
                d="M14.107 12.5l10.56-10.56A1.136 1.136 0 1 0 23.06.333L12.5 10.893 1.94.333A1.136 1.136 0 1 0 .333 1.94l10.56 10.56L.333 23.06a1.136 1.136 0 1 0 1.607 1.607l10.56-10.56 10.56 10.56c.222.222.513.333.804.333a1.136 1.136 0 0 0 .803-1.94L14.107 12.5z"
              />
            </svg>
          </button>
        </Col>
      </Row>
    </div>
  );
};

LoyaltyPromoBanner.propTypes = {
  className: PropTypes.string.isRequired,
  cookieID: PropTypes.string.isRequired,
  richTextList: PropTypes.arrayOf(PropTypes.object),
  dataLocator: PropTypes.string,
};

LoyaltyPromoBanner.defaultProps = {
  richTextList: [
    {
      richText: { text: '' },
      link: {},
    },
  ],
  dataLocator: '',
};

export default withStyles(LoyaltyPromoBanner, style);
export { LoyaltyPromoBanner as LoyaltyPromoBannerVanilla };
