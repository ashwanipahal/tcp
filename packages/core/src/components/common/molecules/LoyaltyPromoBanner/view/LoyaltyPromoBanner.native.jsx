import React from 'react';
import { View } from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import PropTypes from 'prop-types';
import { RichText, Anchor } from '../../../atoms';
import { Wrapper, StyledImage, Touchable } from '../LoyaltyPromoBanner.style.native';

const crossImage = require('../../../../../../src/assets/close.png');

const cookieName = '@mprAboveHead_-1002:key';
const date = new Date();
const currentDateValue = date.getTime();
const daysAlive = 10;

class LoyaltyPromoBanner extends React.PureComponent {
  /**
   * To manage the state of icons on the
   * basis of visible & hide .
   */
  constructor(props) {
    super(props);
    this.state = {
      bannerClosed: false,
    };
  }

  componentDidMount() {
    this.isDisplay();
  }

  /**
   * To save the date
   */
  setDate = () => {
    const expiteDateValue = date.setTime(date.getTime() + daysAlive * 24 * 60 * 60 * 1000);
    try {
      AsyncStorage.setItem(cookieName, expiteDateValue.toString());
    } catch (error) {
      // eslint-disable-next-line rule
      console.info('error', error);
    }
  };

  /**
   * To fetch the date
   */
  retrieveDate = async () => {
    try {
      return await AsyncStorage.getItem(cookieName);
    } catch (error) {
      // Error retrieving data
      return null;
    }
  };

  /**
   * To check the visiblity of the banner
   */
  isDisplay = () => {
    this.retrieveDate().then(data => {
      if (data !== null) {
        if (currentDateValue <= parseInt(data, 10)) {
          this.setState({
            bannerClosed: true,
          });
        } else {
          this.setState({
            bannerClosed: false,
          });
        }
      }
    });
  };

  /**
   * This function validate the iconView.
   */
  validateView = () => {
    this.setDate();
    this.setState({ bannerClosed: true });
  };

  /**
   * To render the view
   */
  render() {
    const {
      richTextList: [{ richText, link }],
      navigation,
    } = this.props;

    const { bannerClosed } = this.state;

    return (
      <View>
        {bannerClosed ? (
          <View />
        ) : (
          <Anchor navigation={navigation} url={link.url}>
            <Wrapper>
              <RichText
                source={{
                  html: richText.text,
                }}
              />
              <Touchable
                accessibilityRole="button"
                accessibilityLabels="close"
                onPress={this.validateView}
              >
                <StyledImage source={crossImage} />
              </Touchable>
            </Wrapper>
          </Anchor>
        )}
      </View>
    );
  }
}

LoyaltyPromoBanner.propTypes = {
  richTextList: PropTypes.arrayOf(PropTypes.object),
  navigation: PropTypes.shape({}).isRequired,
};

LoyaltyPromoBanner.defaultProps = {
  richTextList: [
    {
      richText: { text: '' },
      link: {},
    },
  ],
};

export default LoyaltyPromoBanner;
export { LoyaltyPromoBanner as LoyaltyPromoBannerVanilla };
