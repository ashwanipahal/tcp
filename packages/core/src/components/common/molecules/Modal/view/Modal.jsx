/** @module Modal
 *  @summary Wrapper component for react-modal.
 *  Accepts modal content as children and
 *  Other properties of modal to function.
 */

import React from 'react';
import ReactModal from 'react-modal';
import PropTypes from 'prop-types';
import Grid from '../../Grid';
import Row from '../../../atoms/Row';
import Col from '../../../atoms/Col';
import ModalHeader from './ModalHeader';
import styles from '../Modal.style';
import Config from '../Modal.config';
import withStyles from '../../../hoc/withStyles';

class Modal extends React.PureComponent {
  componentDidMount() {
    // Make sure to bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
    const rootElemId = '__next';
    if (process.env.NODE_ENV !== 'test' && document.getElementById(rootElemId)) {
      ReactModal.setAppElement(`#${rootElemId}`);
    }
  }

  render() {
    const { children, ...otherProps } = this.props;
    const {
      colSet,
      onRequestClose,
      title,
      heading,
      fixedWidth,
      className,
      dataLocator,
      dataLocatorHeader,
      closeIconDataLocator,
      headingStyle,
      closeIconLeftAligned,
      ariaLabelledby,
      ariaDescribedby,
      innerContentClassName = '',
      overlayClassName,
      isOpen,
    } = otherProps;
    const column = colSet || Config.MODAL_COL_DEFAULTS;

    return isOpen ? (
      <ReactModal
        {...otherProps}
        overlayClassName={`${className} ${overlayClassName}`}
        aria={{
          labelledby: ariaLabelledby,
          describedby: ariaDescribedby,
        }}
      >
        {!fixedWidth && (
          <Grid>
            <Row>
              <Col
                colSize={column}
                className={`TCPModal__InnerContent ${innerContentClassName}`}
                data-locator={dataLocator}
              >
                <ModalHeader
                  closeFunc={onRequestClose}
                  title={title}
                  heading={heading}
                  closeIconDataLocator={closeIconDataLocator}
                  closeIconLeftAligned={closeIconLeftAligned}
                  dataLocatorHeader={dataLocatorHeader}
                  headingStyle={headingStyle}
                />
                {children}
              </Col>
            </Row>
          </Grid>
        )}
        {fixedWidth && (
          <div
            className={`TCPModal__InnerContent ${innerContentClassName}`}
            data-locator={dataLocator}
          >
            <ModalHeader
              closeFunc={onRequestClose}
              title={title}
              heading={heading}
              closeIconDataLocator={closeIconDataLocator}
              closeIconLeftAligned={closeIconLeftAligned}
              dataLocatorHeader={dataLocatorHeader}
              headingStyle={headingStyle}
            />
            {children}
          </div>
        )}
      </ReactModal>
    ) : null;
  }
}

Modal.propTypes = {
  children: PropTypes.element.isRequired,
};

// TODO removed errorBoundary from modal as its not working properly right now
export default withStyles(Modal, styles);
export { Modal as ModalVanilla };
