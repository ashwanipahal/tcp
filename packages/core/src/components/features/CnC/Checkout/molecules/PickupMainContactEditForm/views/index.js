export { default } from './PickupMainContactEditForm.view';
