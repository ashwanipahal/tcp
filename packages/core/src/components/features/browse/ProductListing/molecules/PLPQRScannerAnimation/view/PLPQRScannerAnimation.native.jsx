import React from 'react';
import { PropTypes } from 'prop-types';
import { Dimensions } from 'react-native';
import ModalNative from '@tcp/core/src/components/common/molecules/Modal';
import FastImageNative from '../../../../../../../../../mobileapp/src/components/common/molecules/FastImage';

const closeIcon = require('@tcp/core/src/assets/close.png');

/**
 *
 * @param {*} navigation
 * @param {*} closeModal
 *
 * closeIconAction calls when user click on close icon of modal
 * navigation is used to take user to back on scanner page while
 * closeModal close the modal.
 *
 */
const closeIconAction = (navigation, closeModal) => {
  if (navigation) {
    navigation.navigate('QRScanner');
  }
  if (closeModal) {
    closeModal();
  }
};

const dimensions = Dimensions.get('screen');

/**
 * Fast image not render style component. Hence passing as style object
 */
const fastImageStyle = {
  width: dimensions.width - 28,
  height: dimensions.height - 290,
  marginTop: 100,
  marginRight: 'auto',
  marginBottom: 100,
  marginLeft: 'auto',
};

/**
 *
 * @param {*} url
 * @param {*} navigation
 * @param {*} resetCustomLoader
 * @param {*} isOpen
 *
 * This function render Given image URL in modal.
 * When user click on close icon it will take user to QRScanner page.
 */
const PLPQRScannerAnimation = ({ url, navigation, resetCustomLoader, isOpen }) => {
  return (
    <ModalNative
      rightAlignCrossIcon
      isOpen={isOpen}
      fullWidth
      horizontalBar={false}
      onRequestClose={() => closeIconAction(navigation, resetCustomLoader)}
      stickyCloseIcon
    >
      <FastImageNative url={url} style={fastImageStyle} />
    </ModalNative>
  );
};

PLPQRScannerAnimation.propTypes = {
  url: PropTypes.string,
  navigation: PropTypes.shape({}),
  resetCustomLoader: PropTypes.func,
  isOpen: PropTypes.bool,
};

PLPQRScannerAnimation.defaultProps = {
  url: '',
  navigation: {},
  resetCustomLoader: () => {},
  isOpen: false,
};

export default PLPQRScannerAnimation;
