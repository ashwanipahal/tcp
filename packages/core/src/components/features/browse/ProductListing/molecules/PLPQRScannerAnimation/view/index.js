export { default } from './PLPQRScannerAnimation.native';
