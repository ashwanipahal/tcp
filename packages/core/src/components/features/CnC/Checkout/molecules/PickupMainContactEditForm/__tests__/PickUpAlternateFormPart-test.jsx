import React from 'react';
import { shallow } from 'enzyme';
import { PickupMainContactEditFormVanilla } from '../views/PickupMainContactEditForm.view';

describe('Coupon component', () => {
  it('should renders correctly when Coupon are not present', () => {
    const props = {
      labels: {},
      className: '',
      isCondensed: false,
      showNoteOnToggle: false,
      isAlternateUpdateChecked: false,
      isMobile: false,
      formData: {},
      isEditing: false,
      isReset: false,
      onClose: jest.fn(),
    };
    const component = shallow(<PickupMainContactEditFormVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
