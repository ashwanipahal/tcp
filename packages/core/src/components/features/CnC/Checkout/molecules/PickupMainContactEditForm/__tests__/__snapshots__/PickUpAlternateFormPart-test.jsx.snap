// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon component should renders correctly when Coupon are not present 1`] = `
<div
  className=""
>
  <div
    className="header"
  >
    <Styled(BodyCopy)
      className={null}
      color="text.primary"
      component="p"
      fontFamily="primary"
      fontSize="fs28"
      fontWeight="regular"
      letterSpacing="normal"
      lineHeight="normal"
      textAlign="left"
    />
    <div
      className="EditAnchor"
    >
      <Styled(Anchor)
        anchorVariation="secondary"
        className="couponModal_print_anchortext"
        dataLocator="pickup-pickupContact-edit-anchor"
        fontSize="fs12"
        noLink={true}
        onClick={[Function]}
        underline={true}
      />
    </div>
  </div>
  <Styled(PickUpContactDisplay)
    formData={Object {}}
  />
</div>
`;
